### DISCUSSION TITLE
Detail texture scale using UV2 doesn't work, any workaround?

### DISCUSSION THREAD

**Discussion by elvisish** [godotengine/godot-proposals#5296] [3D]
I'm trying to come up with a solution to using a detail texture to add a dark overlay to specific sides of a cube, but UV2 mapping is broken with detail textures as far as I can tell. I've made a test project: https://github.com/godotengine/godot/issues/54644#issuecomment-1226127191

Using a transparent next_pass that has a shadow texture wrapped using UV1, I can make an unshaded box look like this:
![image](https://user-images.githubusercontent.com/16231628/187665781-4d2f80ce-218a-463d-a440-87b3ddec0e0b.png)

However, this is inefficient since next_pass is complete second render of the object with a transparent stage. I'd prefer to use a detail texture that is UV mapped the same as the main texture, however there is no way currently of doing this as UV2 does not allow for the same adjustment as UV1 does:
![image](https://user-images.githubusercontent.com/16231628/187666115-da018dc3-9bde-4e2a-95b3-fb3926546b58.png)

It would be great if this bug could be fixed, but it seems unlikely due to 3.x coming to the end of it's life. If there's a way of doing this with a shader that's as fast (or faster) than using a detail texture, I'd be really interested.

**Comment by Calinou** [comment]
The issue you're describing here is actually a bug: https://github.com/godotengine/godot/issues/54644#issuecomment-1226127191
It can already be worked around with a custom shader, as that's how detail maps are made in SpatialMaterial. They don't actually rely on any hardcoded features in the base GLSL shaders.

Please continue the discussion about the feature implementation in https://github.com/godotengine/godot-proposals/issues/3499. This discussions platform should only be used for feature discussions that don't have a matching proposal issue (with a concrete implementation).
